lib/elixir/test/elixir/enum_test.exs

```diff
@@ -1234,6 +1238,10 @@ defmodule EnumTest.Range do
     assert Enum.reverse(1..3, [4, 5, 6]) == [3, 2, 1, 4, 5, 6]
     assert Enum.reverse(-3..5, MapSet.new([-3, -2])) == [5, 4, 3, 2, 1, 0, -1, -2, -3, -3, -2]
     assert Enum.reverse(5..5, [5]) == [5, 5]
+
+    assert_raise ArgumentError, fn ->
+      Enum.reverse([1 | 2], [3])
+    end
```

Comment:
I don't know if it is possible to solve since we rely on :lists.reverse/1,2. Passing an improper list to `Enum.reverse/1` raises `FunctionClauseError` and passing it to `Enum.reverse/2` returns `ArgumentErrror`

Reply:
That's totally fine. We don't guarantee compatibility of errors.